### PR TITLE
Fix running Python commands that include paths with spaces

### PR DESCRIPTION
**Pull request by silvanocerza**
This PR fixes running Python commands that include paths with spaces by wrapping python commands and their arguments in quotation marks.

I fixed this only in Python as I noticed this while trying to run `pytest` in Zed. 

Probably this is not the best approach as it doesn't fix other languages too, though I don't know enough about the codebase to fix it like that. I'm not even sure if it's actually feasible right now.

I didn't add tests for this either as I couldn't really understand how to easily to that, I tried to look at other languages but couldn't find one that tests their `ContextProvider` directly.

Release Notes:

- Fix running Python commands that include paths with spaces

### CODE REVIEW

**Comment by silvanocerza** [comment]
@cla-bot check

**Comment by osiewicz** [comment]
I took liberty to do that; 0993c89 introduces a new helper. We can use that instead in this PR.

File: crates/languages/src/python.rs [hunk 1]
```diff
@@ -388,7 +394,7 @@ impl ContextProvider for PythonContextProvider {
                         args: vec![
                             "-m".to_owned(),
                             "pytest".to_owned(),
-                            VariableName::File.template_value(),
+                            format!("\"{}\"", VariableName::File.template_value()),
```
Comment (osiewicz) [inline crates/languages/src/python.rs:397]:
```suggestion
                            VariableName::File.template_value_with_whitespace(),
```

File: crates/languages/src/python.rs [hunk 2]
```diff
@@ -399,7 +405,7 @@ impl ContextProvider for PythonContextProvider {
                         args: vec![
                             "-m".to_owned(),
                             "pytest".to_owned(),
-                            "$ZED_CUSTOM_PYTHON_TEST_TARGET".to_owned(),
+                            "\"$ZED_CUSTOM_PYTHON_TEST_TARGET\"".to_owned(),
```
Comment (osiewicz) [inline crates/languages/src/python.rs:408]:
```suggestion
                            PYTHON_TEST_TARGET_TASK_VARIABLE.template_value_with_whitespace(),
```

File: crates/languages/src/python.rs [hunk 3]
```diff
@@ -369,7 +375,7 @@ impl ContextProvider for PythonContextProvider {
                         args: vec![
                             "-m".to_owned(),
                             "unittest".to_owned(),
-                            "$ZED_CUSTOM_PYTHON_TEST_TARGET".to_owned(),
+                            "\"$ZED_CUSTOM_PYTHON_TEST_TARGET\"".to_owned(),
```
Comment (osiewicz) [inline crates/languages/src/python.rs:378]:
```suggestion
                            PYTHON_TEST_TARGET_TASK_VARIABLE.template_value_with_whitespace()
```

File: crates/languages/src/python.rs [hunk 4]
```diff
@@ -358,7 +364,7 @@ impl ContextProvider for PythonContextProvider {
                         args: vec![
                             "-m".to_owned(),
                             "unittest".to_owned(),
-                            VariableName::File.template_value(),
+                            format!("\"{}\"", VariableName::File.template_value()),
```
Comment (osiewicz) [inline crates/languages/src/python.rs:367]:
```suggestion
                            VariableName::File.template_value_with_whitespace(),
```

File: crates/languages/src/python.rs [hunk 5]
```diff
@@ -336,14 +339,17 @@ impl ContextProvider for PythonContextProvider {
             TaskTemplate {
                 label: "execute selection".to_owned(),
                 command: PYTHON_ACTIVE_TOOLCHAIN_PATH.template_value(),
-                args: vec!["-c".to_owned(), VariableName::SelectedText.template_value()],
+                args: vec![
+                    "-c".to_owned(),
+                    format!("\"{}\"", VariableName::SelectedText.template_value()),
```
Comment (osiewicz) [inline crates/languages/src/python.rs:344]:
```suggestion
                    VariableName::SelectedText.template_value_with_whitespace(),
```

**Comment by osiewicz** [comment]
Let's not worry about other languages in the scope of this PR.

**Comment by silvanocerza** [comment]
Nice, that's indeed a much nicer solution. 💯

**Comment by osiewicz** [comment]
Thanks!

**Pull request merged**